CRM/Contact/BAO/Contact/Utils.php

```diff
@@ -835,6 +835,12 @@ public static function processSharedAddress(&$address) {
         }
         continue;
       }
+      // in the UI, we always have a checkbox for specifying whether to update the current employer or not
+      // but if not checked, the parameter is just not defined in the address array
+      // we need to force a value because of API backward compatibility (implicitely create relationship if no parameter)
+      if (!isset($values['update_current_employer'])) {
+        $values['update_current_employer'] = 0;
```

Comment:
I'm confused here. The comment says default to true, but the code appears to be defaulting to 0.

Reply:
This is not a default value: this function is called after the form submission, when the user have checked or not the box.
Quickbooks doesn't seems to pass the value of the checkboxes that are not checked. It's either 1 or not defined.
It's just a way to force a 0 value when the user didn't check the box (the user really meant a 0).
Or maybe we should not use a checkbox but rather a yes/no to avoid this ?

Reply:
Note that we only need this because of backward compatibility to avoid creating a relationship when a user uncheck the box.

Reply:
This still doesn't seem right. The current behavior in the API is to update the current employer, but now you are changing the api to *not* update the current employer unless explicitly told to do so. So this is still a breaking change, no?
Switching the checkbox to `advcheckbox` will give you an explicit value from the form, then you can change this fallback.

Reply:
Ok I finally get it - this function doesn't get called by the api. Ok!